logicrunner/logicrunner.go

```diff
@@ -232,127 +207,154 @@ func (lr *LogicRunner) Execute(ctx context.Context, inmsg core.SignedMessage) (c
 	if err != nil {
 		return nil, errors.Wrap(err, "Can't create request")
 	}
-	logicContext := core.LogicCallContext{
+
+	es.callContext = &core.LogicCallContext{
 		Caller:  msg.GetCaller(),
+		Callee:  &ref,
 		Request: reqref,
 		Time:    time.Now(), // TODO: probably we should take it from e
 		Pulse:   *lr.pulse(),
 	}
 
 	switch m := msg.(type) {
 	case *message.CallMethod:
-		re, err := lr.executeMethodCall(logicContext, m, vb)
+		re, err := lr.executeMethodCall(es, m, vb)
 		return re, err
 
 	case *message.CallConstructor:
-		re, err := lr.executeConstructorCall(logicContext, m, vb)
+		re, err := lr.executeConstructorCall(es, m, vb)
 		return re, err
 
 	default:
 		panic("Unknown e type")
 	}
 }
 
-func (lr *LogicRunner) pulse() *core.Pulse {
-	pulse, err := lr.Ledger.GetPulseManager().Current()
-	if err != nil {
-		panic(err)
-	}
-	return pulse
-}
-
+// ObjectBody is an inner representation of object and all it accessory
+// make it private again when we start it serialize before sending
 type ObjectBody struct {
-	Object core.ObjectDescriptor
-	Class  core.ClassDescriptor
-	Code   core.CodeDescriptor
+	objDescriptor   core.ObjectDescriptor
+	Object          []byte
```

Comment:
Object's memory is duplicated as Object field and objDescriptor.Memory() method. This is wrong.

Reply:
no, it's a link.
but you can not get object on one machine and update it on another, it is a workaround for ledger